plugin/src/main/kotlin/me/gabber235/typewriter/entry/EntryDatabase.kt

```diff
@@ -44,9 +50,53 @@ object EntryDatabase {
 
 		this.entries = pages?.flatMap { it.entries } ?: listOf()
 
+		this.commandEvents = this.events.filter {
+			it::class.simpleName == "RunCommandEventEntry"
+		}
+
+		registerCommands()
 		EntryListeners.register()
 
-		println("Loaded ${facts.size} fact, ${entities.size} entities, ${events.size} event, ${dialogue.size} dialogue and ${actions.size} action entries")
+		println("Loaded ${facts.size} facts, ${entities.size} entities, ${events.size} events, ${dialogue.size} dialogues, ${actions.size} actions, and ${commandEvents.size} commands.")
+	}
+
+	private fun registerCommands() {
+		// https://www.spigotmc.org/threads/tutorial-how-to-register-unregister-custom-commands-at-runtime.493956/
+		commandEvents.forEach { event ->
+			val aliases = mutableListOf<String>()
+			aliases.add(event.command)
+
+			val usage = "/${event.command}"
+			val description = "description" // TODO: add description to entry
+
+			val command = object : BukkitCommand(event.command, description, usage, aliases) {
+				override fun execute(sender: CommandSender, commandLabel: String, args: Array<out String>?): Boolean {
+					// executes on its own
+					return true
+				}
+			}
+
+			plugin.getCommand(event.command)?.unregister(getCommandMap())
+			getCommandMap().register(plugin.name, command)
+
+			println("Registered command ${event.command} for ${event.name} (${event.id})")
+		}
+	}
+
+	private fun getCommandMap(): CommandMap {
+		var map: CommandMap = Bukkit.getServer().commandMap
+
+		try {
+			val server = Bukkit.getServer()
+			val serverClass = server.javaClass
+			val commandMapField = serverClass.getDeclaredField("commandMap")
+			commandMapField.isAccessible = true
+			map = commandMapField.get(server) as CommandMap
+		} catch (e: Exception) {
+			e.printStackTrace()
+		}
```

Comment:
Maybe move this to a different class is this is not very relevant for the `EntryDatabase`.